rholang/src/main/scala/coop/rchain/rholang/interpreter/RholangCLI.scala

```diff
@@ -122,28 +122,36 @@ object RholangCLI {
     } yield ()
     val evaluatorFuture: CancelableFuture[Unit] = evaluatorTask.runAsync
     keepTrying(evaluatorFuture, store)
+    print("\n> ")
   }
 
   private def printTask(normalizedTerm: Par): Task[Unit] =
     Task {
       print("Evaluating:\n")
       println(PrettyPrinter().buildString(normalizedTerm))
-      print("\n> ")
     }
 
   @tailrec
   private def keepTrying(evaluatorFuture: CancelableFuture[Unit],
                          persistentStore: IStore[Channel, Seq[Channel], Seq[Channel], Par]): Unit =
-    Await.ready(evaluatorFuture, 5.seconds).value match {
-      case Some(Success(_)) =>
-        print("Storage Contents:\n")
-        StoragePrinter.prettyPrint(persistentStore)
-        print("\n> ")
-      case Some(Failure(e: TimeoutException)) =>
+    try {
+      Await.ready(evaluatorFuture, 5.seconds).value match {
+        case Some(Success(_)) =>
+          print("Storage Contents:\n")
+          StoragePrinter.prettyPrint(persistentStore)
+          print("\n> ")
+        case Some(Failure(e)) => {
+          println("Caught boxed exception: " + e)
+          throw e
```

Comment:
I don't think it is necessary to rethrow this exception. Might as well just handle this all in the catch clause.